insert_last_verified.py

```diff
@@ -0,0 +1,180 @@
+import json
+import os
+import subprocess
+import sys
+from datetime import datetime
+
+from bs4 import BeautifulSoup
+
+# Check if the build directory is provided as an argument in the Makefile
+if len(sys.argv) < 2:
+    print("Error: Build directory not provided. Exiting.")
+    exit(1)
+
+build_dir = sys.argv[1]
+print(f"Build directory: {build_dir}")
+
+json_file_path = "tutorials-review-data.json"
+build_dir = "_build/html"  # for testing after _build/html is created
+
+# paths to skip from the post-processing script
+paths_to_skip = [
+    "beginner/examples_autograd/two_layer_net_custom_function",  # not present in the repo
+    "beginner/examples_nn/two_layer_net_module",  # not present in the repo
+    "beginner/examples_tensor/two_layer_net_numpy",  # not present in the repo
+    "beginner/examples_tensor/two_layer_net_tensor",  # not present in the repo
+    "beginner/examples_autograd/two_layer_net_autograd",  # not present in the repo
+    "beginner/examples_nn/two_layer_net_optim",  # not present in the repo
+    "beginner/examples_nn/two_layer_net_nn",  # not present in the repo
+    "intermediate/coding_ddpg",  # not present in the repo - will delete the carryover
+]
+# Mapping of source directories to build directories
+source_to_build_mapping = {
+    "beginner": "beginner_source",
+    "recipes": "recipes_source",
+    "distributed": "distributed",
+    "intermediate": "intermediate_source",
+    "prototype": "prototype_source",
+    "advanced": "advanced_source",
+    "": "",  # root dir for index.rst
+}
+
+
+# Use git log to get the creation date of the file
+def get_creation_date(file_path):
+    try:
+        result = subprocess.run(
+            ["git", "log", "--diff-filter=A", "--format=%aD", "--", file_path],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        if result.stdout:
+            creation_date = result.stdout.splitlines()[0]
+            creation_date = datetime.strptime(creation_date, "%a, %d %b %Y %H:%M:%S %z")
+            formatted_date = creation_date.strftime("%b %d, %Y")
+        else:
+            formatted_date = "Unknown"
+        return formatted_date
+    except subprocess.CalledProcessError:
+        return "Unknown"
```

Comment:
It feels like `get_last_updated` and `get_created` could benefit from some refactoring of common code. 
And why `unknown` is an accepted answer? Shouldn't it raise an exception that should be handled elsewhere?